src/Raven.Server/Documents/Revisions/RevisionsStorage.cs

```diff
@@ -1898,7 +1898,40 @@ public void ForceDeleteAllRevisionsFor(DocumentsOperationContext context, string
                 (table, result) => GetAllRevisions(context, table, prefixSlice, maxDeletesUponUpdate, shouldSkip, result), tombstoneFlags);
         }
 
-        private (bool MoreWork, long Deleted) ForceDeleteAllRevisionsFor(DocumentsOperationContext context, Slice lowerId, Slice prefixSlice, CollectionName collectionName, 
+        private (bool MoreWork, long Deleted) ForceDeleteAllRevisionsFor(DocumentsOperationContext context, Slice lowerId, Slice prefixSlice, CollectionName collectionName, long? maxDeletesUponUpdate,
+            long etagBarrier, DocumentFlags tombstoneFlags = DocumentFlags.None)
+        {
+            return ForceDeleteAllRevisionsFor(context, lowerId, prefixSlice, collectionName, GetRevisions, tombstoneFlags);
+
+            IEnumerable<Document> GetRevisions(Table table, DeleteOldRevisionsResult result)
+            {
+                var revisions = GetAllRevisions(context, table, prefixSlice, maxDeletesUponUpdate, shouldSkip: null, result);
+                foreach (var r in revisions)
+                {
+                    if (context.CanContinueTransaction == false)
```

Comment:
let's put that at the end, so we will get at least one, otherwise in an edge case you will never delete a revision

Reply:
Done